### PR TITLE
Modify ABICallElection

### PR DESCRIPTION
**Pull request by julia-zack**
There is an ongoing effort to organize Bridge classes in their package and avoid circular dependencies.
One of the classes we have identified that could be moved to their package are those related to ABI voting.
To capitalize this effort, there are some improvements that can be done in each class.

This PR modifies the ABICallElection:
- Make getWinner return Optional instead of null. 
- Add logs.
- Make validate method throw a new UnauthorizedVoterException.

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/peg/UnauthorizedVoterException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package co.rsk.peg;
+
+public class UnauthorizedVoterException extends RuntimeException {
```
Comment (marcos-iov) [inline rskj-core/src/main/java/co/rsk/peg/UnauthorizedVoterException.java:3]: Maybe this should go in `abi` package?

File: rskj-core/src/main/java/co/rsk/peg/abi/ABICallElection.java [hunk 2]
```diff
@@ -61,46 +66,54 @@ public void clear() {
      */
     public boolean vote(ABICallSpec callSpec, RskAddress voter) {
```
Comment (marcos-iov) [inline rskj-core/src/main/java/co/rsk/peg/abi/ABICallElection.java:67]: We should probably log the voter also. Maybe at the beginning of the method

File: rskj-core/src/main/java/co/rsk/peg/UnauthorizedVoterException.java (hunk 1 above)
Comment (julia-zack) [reply rskj-core/src/main/java/co/rsk/peg/UnauthorizedVoterException.java:3]: i suggest we create an `/authorizer` package inside `/peg`, to have this class and AdressBasedAuthorizer. 
Wdyt?
This way we avoid the circular dependency between `/peg` and `/peg/abi` too

File: rskj-core/src/main/java/co/rsk/peg/abi/ABICallElection.java (hunk 2 above)
Comment (julia-zack) [reply rskj-core/src/main/java/co/rsk/peg/abi/ABICallElection.java:67]: Done

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**